### ISSUE TITLE
java error with pubsub connector in 1.39

### ISSUE THREAD

**Issue by Marc3001** [zendesk/maxwell#1939]
We got issue after upgrading from 1.38.0 to 1.39.2

```
2022-11-03 19:37:04 INFO  Maxwell - Starting Maxwell. maxMemory: 518979584 bufferMemoryUsage: 0.25
2022-11-03 19:37:07 INFO  TaskManager - Stopping 0 tasks
2022-11-03 19:37:07 INFO  TaskManager - Stopped all tasks
Exception in thread "main" java.util.ServiceConfigurationError: io.grpc.NameResolverProvider: io.grpc.grpclb.SecretGrpclbNameResolverProvider$Provider Unable to get public no-arg constructor
	at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:582)
	at java.base/java.util.ServiceLoader.getConstructor(ServiceLoader.java:673)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1233)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1265)
	at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1300)
	at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1385)
	at io.grpc.ServiceProviders.loadAll(ServiceProviders.java:67)
	at io.grpc.NameResolverRegistry.getDefaultRegistry(NameResolverRegistry.java:101)
	at io.grpc.internal.AbstractManagedChannelImplBuilder.<init>(AbstractManagedChannelImplBuilder.java:114)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder.<init>(NettyChannelBuilder.java:136)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder.<init>(NettyChannelBuilder.java:131)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder.forAddress(NettyChannelBuilder.java:117)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelProvider.builderForAddress(NettyChannelProvider.java:37)
	at io.grpc.netty.shaded.io.grpc.netty.NettyChannelProvider.builderForAddress(NettyChannelProvider.java:23)
	at io.grpc.ManagedChannelBuilder.forAddress(ManagedChannelBuilder.java:39)
	at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.createSingleChannel(InstantiatingGrpcChannelProvider.java:350)
	at com.google.api.gax.grpc.ChannelPool.<init>(ChannelPool.java:105)
	at com.google.api.gax.grpc.ChannelPool.create(ChannelPool.java:83)
	at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.createChannel(InstantiatingGrpcChannelProvider.java:236)
	at com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.getTransportChannel(InstantiatingGrpcChannelProvider.java:230)
	at com.google.api.gax.rpc.ClientContext.create(ClientContext.java:201)
	at com.google.cloud.pubsub.v1.stub.GrpcPublisherStub.create(GrpcPublisherStub.java:203)
	at com.google.cloud.pubsub.v1.Publisher.<init>(Publisher.java:189)
	at com.google.cloud.pubsub.v1.Publisher.<init>(Publisher.java:89)
	at com.google.cloud.pubsub.v1.Publisher$Builder.build(Publisher.java:838)
	at com.zendesk.maxwell.producer.MaxwellPubsubProducerWorker.<init>(MaxwellPubsubProducer.java:192)
	at com.zendesk.maxwell.producer.MaxwellPubsubProducer.<init>(MaxwellPubsubProducer.java:111)
	at com.zendesk.maxwell.MaxwellContext.getProducer(MaxwellContext.java:538)
	at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:257)
	at com.zendesk.maxwell.Maxwell.start(Maxwell.java:227)
	at com.zendesk.maxwell.Maxwell.main(Maxwell.java:337)
Caused by: java.lang.VerifyError: Cannot inherit from final class
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3137)
	at java.base/java.lang.Class.getConstructor0(Class.java:3342)
	at java.base/java.lang.Class.getConstructor(Class.java:2151)
	at java.base/java.util.ServiceLoader$1.run(ServiceLoader.java:660)
	at java.base/java.util.ServiceLoader$1.run(ServiceLoader.java:657)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.ServiceLoader.getConstructor(ServiceLoader.java:668)
	... 29 more
```

We're using maxwell to read from MysQL and to push to GCP pubsub.
Our maxwell process is hosted on k8s and using docker.io official images.

**Comment by mozarik** [comment]
Same as me but mine is  `Cannot parse argument '1.0' of option pubsub_rpc_timeout_multiplier`

```
joptsimple.OptionArgumentConversionException: Cannot parse argument '1.0' of option pubsub_rpc_timeout_multiplier
	at joptsimple.AbstractOptionSpec.convertWith(AbstractOptionSpec.java:92)
	at joptsimple.ArgumentAcceptingOptionSpec.convert(ArgumentAcceptingOptionSpec.java:277)
	at joptsimple.OptionSet.valuesOf(OptionSet.java:223)
	at joptsimple.OptionSet.valueOf(OptionSet.java:172)
	at joptsimple.OptionSet.valueOf(OptionSet.java:153)
	at com.zendesk.maxwell.util.AbstractConfig.fetchOption(AbstractConfig.java:132)
	at com.zendesk.maxwell.util.AbstractConfig.fetchStringOption(AbstractConfig.java:140)
	at com.zendesk.maxwell.MaxwellConfig.setup(MaxwellConfig.java:1080)
	at com.zendesk.maxwell.MaxwellConfig.parse(MaxwellConfig.java:1031)
	at com.zendesk.maxwell.MaxwellConfig.<init>(MaxwellConfig.java:655)
	at com.zendesk.maxwell.Maxwell.main(Maxwell.java:316)
Caused by: joptsimple.internal.ReflectionException: java.lang.NumberFormatException: For input string: "1.0"
	at joptsimple.internal.Reflection.reflectionException(Reflection.java:136)
	at joptsimple.internal.Reflection.invoke(Reflection.java:118)
	at joptsimple.internal.MethodInvokingValueConverter.convert(MethodInvokingValueConverter.java:48)
	at joptsimple.internal.Reflection.convertWith(Reflection.java:124)
	at joptsimple.AbstractOptionSpec.convertWith(AbstractOptionSpec.java:90)
	... 10 more
Caused by: java.lang.NumberFormatException: For input string: "1.0"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Long.parseLong(Long.java:692)
	at java.base/java.lang.Long.valueOf(Long.java:1144)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at joptsimple.internal.Reflection.invoke(Reflection.java:116)
```

**Comment by mozarik** [comment]
just fixed this error in here PR https://github.com/zendesk/maxwell/pull/1940

> Cannot parse argument '1.0' of option pubsub_rpc_timeout_multiplier

**Comment by osheroff** [comment]
I'm in the dark here without a working pubsub setup.  can you try:
https://github.com/zendesk/maxwell/tree/ben/try_upgrade_gcps

**Comment by mozarik** [comment]
@osheroff let me see if this work. currently got this issue too

**Comment by mozarik** [comment]
> I'm in the dark here without a working pubsub setup. can you try:

upgrading it wont work still have issue

**Comment by mozarik** [comment]
Do you think we need this to enforce upper bound dependency ? 

```
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-enforcer-plugin</artifactId>
        <version>1.4.1</version>
        <executions>
          <execution>
            <id>enforce</id>
            <goals>
              <goal>enforce</goal>
            </goals>
            <configuration>
              <rules>
                <requireUpperBoundDeps/>
              </rules>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

**Comment by osheroff** [comment]
try again; I moved some deps around which changed some transitive versions.

**Comment by mozarik** [comment]
@osheroff ok let me try it

**Comment by osheroff** [comment]
fixed this in 1.39.3
